lms/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -3,9 +3,37 @@
 """
 from __future__ import unicode_literals
 
+import time
+from datetime import datetime
+from pytz import utc
+
 from django.contrib.auth import authenticate, get_user_model
 from oauth2_provider.oauth2_validators import OAuth2Validator
 
+from oauth2_provider.models import AccessToken
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from django.conf import settings
+
+from .models import RestrictedApplication
+
+
+@receiver(pre_save, sender=AccessToken)
+def on_access_token_presave(sender, instance, *args, **kwargs):
+    """
+    A hook on the AccessToken. For when we grant authorization-codes,
+    since we do not have protected scopes, we must mark all
+    AcessTokens as expired for 'restricted applications'.
+
+    We do this as a pre-save hook on the ORM
+    """
+
+    is_application_restricted = RestrictedApplication.objects.filter(application=instance.application).exists()
+
+    if is_application_restricted:
+        # put the expire timestamp into the beginning of the epoch
+        instance.expires = datetime(1970, 1, 1, tzinfo=utc)
```

Comment:
@clintonb I tried this but I'm getting access_tokens rows in the database with the expiry year of '0001-01-01' which seems a bit abnormal and I'm concerned certain clients might not properly parse this.

Any issue on keeping this to be 1970-01-01 which is - as I understand - the beginning of the Unix epoch?